turbo/rpchelper/rpc_block.go

```diff
@@ -0,0 +1,51 @@
+package rpchelper
+
+import (
+	"fmt"
+
+	"github.com/ledgerwatch/erigon-lib/kv"
+	"github.com/ledgerwatch/erigon/common"
+	"github.com/ledgerwatch/erigon/core/rawdb"
+	"github.com/ledgerwatch/erigon/eth/stagedsync/stages"
+)
+
+func GetLatestBlockNumber(tx kv.Tx) (uint64, error) {
+	forkchoiceHeadHash := rawdb.ReadForkchoiceHead(tx)
+	if forkchoiceHeadHash != (common.Hash{}) {
+		forkchoiceHeadNum := rawdb.ReadHeaderNumber(tx, forkchoiceHeadHash)
+		if forkchoiceHeadNum != nil {
+			return *forkchoiceHeadNum, nil
+		}
+	}
+
+	blockNum, err := stages.GetStageProgress(tx, stages.Execution)
+	if err != nil {
+		return 0, fmt.Errorf("getting latest block number: %w", err)
+	}
+
+	return blockNum, nil
+}
+
+func GetFinalizedBlockNumber(tx kv.Tx) (uint64, error) {
+	forkchoiceFinalizedHash := rawdb.ReadForkchoiceFinalized(tx)
+	if forkchoiceFinalizedHash != (common.Hash{}) {
+		forkchoiceFinalizedNum := rawdb.ReadHeaderNumber(tx, forkchoiceFinalizedHash)
+		if forkchoiceFinalizedNum != nil {
+			return *forkchoiceFinalizedNum, nil
+		}
+	}
+
+	return 0, nil
```

Comment:
Per [the spec](https://github.com/ethereum/execution-apis/pull/200): "Before the merge transition is finalized, any call querying for `finalized` or `safe` block MUST be responded to with `-39001: Unknown block` error".

So we should return an `rpc.CustomError` here.